### DISCUSSION TITLE
the View Has been Viewed.Why my Customer Control Is Null(like PathRolateExtHelper)

### DISCUSSION THREAD

**Discussion by hubing53** [AvaloniaUI/Avalonia#9018] [Q&A] [answered]
![1664091173270_a1b36403](https://user-images.githubusercontent.com/112156001/192133224-d65cba92-b1a1-4ca5-8561-37e282ef5b64.png)
![1664091071786_b258464f](https://user-images.githubusercontent.com/112156001/192133235-57aad722-40a6-4cb5-ae34-3a5a59c82351.png)

**Comment by timunie** [comment, marked as answer]
If you are trying to reference a control by it's name in xaml, you have to use 

```cs
this.Find("MyName");
```

If that wasn't your issue, we need a sample App to check on our own.

**Reply by hubing53** [reply]
My Direction Property still change But the View Binding not change
![1664098931951_9e4b1571](https://user-images.githubusercontent.com/112156001/192137451-42992af3-376f-4351-b03b-f20d50691608.png)
![1664099046705_18f14c43](https://user-images.githubusercontent.com/112156001/192137478-197bfe2b-cc00-4c22-ba2e-36adfbfa2b27.png)

**Reply by timunie** [reply]
In that case I see two options:

1. Have a look at AwesomeAvalonia and find a lib that maybe has a control similar to yours. See the source code there. https://github.com/AvaloniaCommunity/awesome-avalonia

2. Provide a minimal sample, so we may send you a fix if easily possible.

**Reply by hubing53** [reply]
i test the derectionproperty has been set on ArcSegment  But The View Not Viewed
![1664107745988_3509aece](https://user-images.githubusercontent.com/112156001/192142686-e8fc5364-6c5f-4d3b-b06c-2bfbd1a4211c.png)

**Reply by maxkatz6** [reply]
I don't think neither PathSegment not ArcSegment do force renderer to update on their property changed. It should be changed in 11.0.

**Comment by hubing53** [comment]
![1664109085473_b1b26deb](https://user-images.githubusercontent.com/112156001/192143515-f120025b-61e5-4146-b474-b86caa758be5.png)
I Think The UI Not Fresh

**Reply by hubing53** [reply]
I use Code Test  if the ArcSegment in the path than you only change the Arcsegment point  the path not changeView
![1664110244260_37ef35c6](https://user-images.githubusercontent.com/112156001/192144430-9ed0c241-4c30-4738-9044-83fa43ca7793.png)

![1664110381598_b6ac3c06](https://user-images.githubusercontent.com/112156001/192144465-a1333f9e-4374-4013-9f76-262eabf6bafa.png)
 I should change Another Method  to Achievement
